### ISSUE TITLE
Exception while handling ionos API error

### ISSUE THREAD

**Issue by JulienPalard** [helgeerbe/certbot-dns-ionos#24]
In dns_ionos.py line 125 there's:

```python
        if resp.status_code != 200:
            content = json.loads(resp.content)[0] # on error content is array with 1 element                                                       
```

But I got a IONOS API answer with no array, just a message, like this:

    {"message":"Missing or invalid API key."}

Maybe just using `error_msg = resp.reason + " " + resp.text` would be more robust? I won't care reading a bit of json in my logs :D

**Comment by DorianCoding** [comment]
Hello,

At least you know your API key is invalid no?

**Comment by DorianCoding** [comment]
Hello,

I think it is more clear now, can you tell me?

**Comment by helgeerbe** [comment]
Should be fixed with 2024.10.19
